src/app/core/services/maprules.service.ts

```diff
@@ -47,39 +48,56 @@ export class MapRulesService {
     }
   }
 
-  isEmpty(empty: any) {
-      return empty === '';
+  saveForm(value: {[name: string]: any}, presetGeometries: any[]){
+    const scrubbedForm = this.serialize(value, presetGeometries);
+    return this.saveNewConfig(scrubbedForm);
   }
 
-  removeEmpty(config: {[name: string]: any}){
-    var next = {}
-    var $scope = this;
-    Object.keys(config).forEach((key) => {
-      if (Array.isArray(config[key])) {
-        var nextArray = [];
-        config[key].forEach(subVal => {
-          if (subVal instanceof Object) {
-            const flush = $scope.removeEmpty(subVal);
-            if (Object.keys(flush).length > 0) {
-              nextArray.push(flush);
+  saveNewConfig(scrubbedForm: {[name: string]: any}){
+    return this.http.post(this.mapRulesUrl, scrubbedForm, this.httpOptions).pipe(
+      catchError(this.handleError)
+    );
+  }
+
+  serialize(config: any, presetGeometries: any[]): any {
+    return {
+      name: config.mapruleName,
+      presets: config.presets.map(function (preset, index) {
+        return {
+          name: preset.presetName,
+          geometry: presetGeometries[index],
+          primary: preset.primary.map(function (primary) {
+            return {
+              key: primary.primaryKey,
+              val: primary.primaryVal
             }
-          } else {
-            if (!$scope.isEmpty(nextArray)) {
-              nextArray.push(subVal);
+          }),
+          fields: preset.fields.map(function (field) {
+            return {
+              key: field.fieldKey,
+              keyCondition: FieldConfigService.KEY_CONDITIONS.indexOf(field.fieldKeyCondition),
+              values: !field.fieldVal.length ? [] : [{
+                valCondition: FieldConfigService.VAL_CONDITIONS.indexOf(field.fieldValCondition),
+                values: field.fieldVal.length ? field.fieldVal.split(',') : []
+              }]
             }
-          }
-        })
-        next[key] = nextArray;
-      } else if (!$scope.isEmpty(config[key])) {
-          next[key] = config[key]
-      }
-    })
-    return next;
+          }),
+        }
+      }),
+      disabledFeatures: config.disabledFeatures.map(function (disabledFeature) {
+        return {
+          key: disabledFeature.disabledKey,
+          val: disabledFeature.disabledVal.length ? disabledFeature.disabledVal.split(',') : []
+        }
+      })
+    }
   }
 
-
   getMapRule(configId: string){
-    return this.http.get(this.mapRulesUrl + "/" + configId).pipe(catchError(this.handleError));
+    return this.http.get(this.mapRulesUrl + "/" + configId).pipe(
+      tap((maprule: any) => this.currentMapRule = maprule),
```

Comment:
I think I had it in my head that knowing the source maprule config would be helpful, but never actually made use of it. we can probably pluck and if we find a use bring back in the future.